ruby_event_store/lib/ruby_event_store/specification.rb

```diff
@@ -31,13 +31,14 @@ def backward?
 
     attr_reader :result
 
-    def initialize(repository, result = Result.new(:forward, :head, NO_LIMIT, Stream.new(GLOBAL_STREAM)))
+    def initialize(repository, mapper, result = Result.new(:forward, :head, NO_LIMIT, Stream.new(GLOBAL_STREAM)))
+      @mapper = mapper
       @repository  = repository
       @result = result
     end
 
     def stream(stream_name)
-      Specification.new(repository, result.dup.tap { |r| r.stream = Stream.new(stream_name) })
+      Specification.new(repository, mapper, result.dup.tap { |r| r.stream = Stream.new(stream_name) })
```

Comment:
Maybe `result.dup.tap { |r| r.stream = Stream.new(stream_name)` (and similar) should be extracted to:

`result.with_stream(steam_name)` where implementation would be

```
def with_stream(stream_name)
  self.dup.tap { |r| r.stream = Stream.new(stream_name)
end
```